aiida_testing/export_cache/_fixtures.py

```diff
@@ -0,0 +1,375 @@
+        """
+        ignored = list(self._hash_ignored_attributes)  # pylint: disable=protected-access
+        ignored.append('version')
+        self._hash_ignored_attributes = tuple(ignored)  # pylint: disable=protected-access
+
+        objects = [
+            #importlib.import_module(self.__module__.split('.', 1)[0]).__version__,
+            {
+                key: val
+                for key, val in self.attributes_items() if key not in self._hash_ignored_attributes
+                and key not in self._updatable_attributes
+            },
+            #self._repository._get_base_folder(),
+            #self.computer.uuid if self.computer is not None else None
+        ]
+
+        return objects
+
+    # since we still want versioning for plugin datatypes and calcs we only monkeypatch aiida datatypes
+    classes_to_patch = [Dict, SinglefileData, List, FolderData, RemoteData]
+    for classe in classes_to_patch:
+        monkeypatch.setattr(classe, "_get_objects_to_hash", mock_objects_to_hash)
+
+    #BaseData, List, Array, ...
+
+
+@pytest.fixture(scope='function')
+def run_with_cache(export_cache, load_cache):
+    """
+    Fixture to automatically import an aiida graph for a given process builder.
+    """
+    def _run_with_cache(
+        builder: dict,  #aiida process builder class, or dict, if process class is given
+        process_class=None,
+        label: str = '',
+        data_dir: ty.Union[str, pathlib.Path] = 'data_dir',
+    ):
+        """
+        Function, which checks if a aiida export for a given Process builder exists,
+        if it does it imports the aiida graph and runs the builder with caching.
+        If the cache does not exists, it still runs the builder but creates an
+        export afterwards.
+
+        Inputs:
+
+        builder : AiiDA Process builder class,
+        data_dir: optional
+            Absolute path of the directory where the exported workchain graphs are
+            stored.
+
+        ignore_nodes : list string, ignore input nodes with these labels/link labels to ignore in hash.
+        # needed?
+        """
+
+        cache_exists = False
+        if data_dir == 'data_dir':
+            cwd = pathlib.Path(os.getcwd())  # Might be not the best idea.
+            data_dir = (cwd / 'data_dir')  # TODO: get from config?
+
+        bui_hash, input_nodes = get_hash_process(builder)
+
+        if process_class is None:  # and isinstance(builder, dict):
+            process_class = builder.process_class  # type: ignore
+            # we assume ProcessBuilder, since type(ProcessBuilder) is abc
+        #else:
+        #    raise TypeError(
+        #        'builder has to be of type ProcessBuilder if no process_class is specified'
+        #    )
+        name = label + str(process_class).split('.')[-1].strip("'>") + '-nodes-' + bui_hash
+        print(name)
+
+        # check existence
+        full_import_path = str(data_dir) + '/' + name + '.tar.gz'
```

Comment:
The hard-coded slash could be problematic: To be platform-agnostic, this should be 

```python
cache_path = pathlib.Path(data_dir) / (name + '.tar.gz')
```